src/backend/joanie/payment/models.py

```diff
@@ -1 +1,463 @@
+        """
+        # - Compute the balance
+        return self.transactions_balance - self.invoiced_balance
+
+    def _update_context(self):
+        """
+        Update or create the invoice context for all languages.
+
+        In order to save space, we want to generate related document only on
+        demand. Furthermore, we need to store product description to prevent
+        inconsistency in case of product would be updated. That's why we store
+        a product information for all active language into the invoice, then we
+        are able to generate invoice document with consistent data only on
+        demand.
+
+        Saving is left to the caller.
+        """
+
+        context = {}
+        related_product = self.get_related_order().product
+        for language, __ in settings.LANGUAGES:
+            with switch_language(related_product, language):
+                context[language] = {
+                    "order": {
+                        "product": {
+                            "name": related_product.title,  # pylint: disable=no-member
+                            "description": related_product.description,  # noqa pylint: disable=no-member,line-too-long
+                        }
+                    }
+                }
+
+        self.localized_context = context
+
+    def get_document_context(self, language_code=None):
+        """
+        Build the invoice document context for the given language.
+        If no language_code is provided, we use the active language.
+        """
+        language_code = language_code or get_language()
+
+        vat = D(settings.JOANIE_VAT)
+        total_amount = self.amount.amount  # pylint: disable=no-member
+        vat_amount = total_amount * vat / 100
+        net_amount = total_amount - vat_amount
+        currency = get_currency_symbol(
+            self.amount.currency.code  # pylint: disable=no-member
+        )
+
+        base_context = {
+            "metadata": {
+                "issued_on": str(self.updated_on),
+                "reference": self.reference,
+                "type": self.type,
+            },
+            "order": {
+                "amount": {
+                    "currency": currency,
+                    "subtotal": str(net_amount),
+                    "total": str(total_amount),
+                    "vat_amount": str(vat_amount),
+                    "vat": str(vat),
+                },
+                "company": settings.JOANIE_INVOICE_COMPANY_CONTEXT,
+                "customer": {
+                    "address": self.recipient_address,
+                    "name": self.recipient_name,
+                },
+                "seller": {
+                    "address": settings.JOANIE_INVOICE_SELLER_ADDRESS,
+                },
+            },
+        }
+
+        try:
+            localized_context = self.localized_context[language_code]
+        except KeyError:
+            fallback_language = settings.PARLER_DEFAULT_LANGUAGE_CODE
```

Comment:
I think in parler each language has a different fallback so there should be some methods to get the default for a language.

Reply:
Yes it's possible by using `LanguageSettings` util. 👍